File: 0058-length-of-last-word/0058-length-of-last-word.cpp
class Solution {
public:
    int lengthOfLastWord(string s) {
        if(s==" "){
            return 0;
        }
        s = " "+s;
        int l = s.size();
        int pR = l-1;
        if(s.at(l-1) == ' '){
            pR--;
            while(s.at(pR) == ' '){
                pR--;
            }
        }
        int count = 0;
        while(s.at(pR) != ' '){
            pR--;
            count++;
        }
        return count;
    }
};

File: 0058-length-of-last-word/0058-length-of-last-word.java
class Solution {
    public int lengthOfLastWord(String s) {
        
        int length = 0;
        int i = s.length()-1;
        s = s+" ";
                while(s.charAt(i)==' '){
                    i--;
                }
                while(s.charAt(i)!=' ' && i>0){
                    length++;
                    i--;
                }
                if(s.charAt(i)!=' '){
                    length++;
                }
        return length;
        
    }
}